java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -996,56 +1002,59 @@ private void handleImageBuildData(ServerAction serverAction, JsonElement jsonRes
+                                .collectKiwiImage(minionServer, bundle.getFilepath(), targetPath)
+                                .orElseThrow(() -> new RuntimeException("Failed to download image."));
+
+                            if (collectResult.getReturnCode() != 0) {
+                                 serverAction.setStatus(ActionFactory.STATUS_FAILED);
+                                 serverAction.setResultMsg(StringUtils
+                                    .left(printStdMessages(collectResult.getStderr(), collectResult.getStdout()),
                                         1024));
-                        }
-                    });
+                            }
+                            else {
+                                ImageFile bundleFile = new ImageFile();
+                                bundleFile.setFile(bundle.getFilename());
+                                bundleFile.setType("bundle");
+                                bundleFile.setImageInfo(info);
+                                info.getImageFiles().add(bundleFile);
+                            }
+                        });
                     });
                 });
-                ImageInspectAction iAction = ActionManager.scheduleImageInspect(
-                        action.getSchedulerUser(),
-                        action.getServerActions()
-                                .stream()
-                                .map(ServerAction::getServerId)
-                                .collect(Collectors.toList()),
-                        Optional.of(action.getId()),
-                        details.getVersion(),
-                        p.getLabel(),
-                        p.getTargetStore(),
-                        Date.from(Instant.now())
-                );
-                try {
-                    TASKOMATIC_API.scheduleActionExecution(iAction);
-                }
-                catch (TaskomaticApiException e) {
-                    LOG.error("Could not schedule image inspection ", e);
-                }
-                infoOpt.ifPresent(info -> {
-                    info.setRevisionNumber(info.getRevisionNumber() + 1);
-                    info.setInspectAction(iAction);
-                    ImageInfoFactory.save(info);
-                });
-        }, () -> LOG.warn("Could not find any profile for profile ID " + imageProfileId));
+            }, () -> LOG.warn("Could not find any profile for profile ID " + imageProfileId));
         }
+        if (serverAction.getStatus().equals(ActionFactory.STATUS_COMPLETED)) {
+            // both building and uploading results succeeded
+            ImageInfoFactory.updateRevision(info);
+            if (info.getImageType().equals(ImageProfile.TYPE_DOCKERFILE)) {
+                ImageInfoFactory.obsoletePreviousRevisions(info);
+            }
+            info.setBuilt(true);
+
+            try {
+                ImageInfoFactory.scheduleInspect(info, Date.from(Instant.now()), action.getSchedulerUser());
```

Comment:
since the image is save later, and commit done even latter, do we have any concurrence problem of tasko starting before the info in available and updated in the database?
You can test it by putting a breakpoint in the ImageInfoFactory.save in this method, and a break point in the taskomatic task.

Reply:
ImageInfoFactory.scheduleInspect calls save at the right point.
The save at the end of handleImageBuildData is only for saving after error.